Mapsui/Extensions/StringExtensions.cs

```
@@ -0,0 +1,38 @@
+﻿using System;
+
+public static class StringExtensions
+{
+    public static string? UriScheme(this string? url)
+    {
+        if (string.IsNullOrEmpty(url))
+        {
+            return null;
+        }
+
+        if (Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out var uri))
+        {
+            return uri.Scheme;
+        }
+
+        return null;
+    }
+
+    public static string? AssureUriScheme(this string? url, string? scheme)
+    {
+        if (string.IsNullOrEmpty(scheme) || string.IsNullOrEmpty(url))
```

Here too I think we should throw instead of return null. Also don't accept null instead there is a good reason for it.

---

i throw now exceptions